changelog/transition-complex.dd

```diff
@@ -0,0 +1,27 @@
+Complex and imaginary types are now deprecated
+
+D currently supports complex and imaginary versions of all floating point types.
+---
+float a = 2;
+ifloat b = 4i;
+cfloat c = a + b;
+assert(c == 2 + 4i);
+---
+
+These types are too specialized to be a part of the core language, and have
+been transitioned over to library types in `std.complex`.
+
+The transition phase for complex and imaginary types has finished, and
+deprecation warnings are now turned on by default.
+
+The deprecation instructs the user to use $(LINK2 https://dlang.org/phobos/std_complex.html#Complex, `std.complex.Complex`) instead.
+---
+import std.complex;
+float a = 2;
+float b = 4;
+auto c = complex(a, b);
+assert(c == complex(2, 4));
+---
+
+The `-transition=complex` switch that previously turn on deprecation warnings
+no longer has an effect and has also been deprecated.
```

Comment:
Well it doesn't trigger a deprecation does it ? I would just leave it as do-nothing, it's usually smoother this way.

Reply:
Also, provided you mention `-transition` earlier, I would add a note that the C compatibility and performances are on par with the native types, which was Ilya's main concern IIRC, and probably will be the concern of anyone using them.

Reply:
Yes, it does. 

https://github.com/dlang/dmd/blob/64663f867ffbcba95b8af0c3745dab5d5ea2aeb0/src/dmd/mars.d#L1596